src/azure-cli/azure/cli/command_modules/profile/__init__.py

```diff
@@ -36,7 +47,6 @@ def load_command_table(self, args):
             g.command('set', 'set_active_subscription')
             g.show_command('show', 'show_subscription')
             g.command('clear', 'account_clear')
-            g.command('list-locations', 'list_locations')
```

Comment:
Even though I wasn't the creator of this `az account list-locations` command which was introduced by https://github.com/Azure/azure-cli/pull/438, I guess it was added under `az account` solely for convenience.

However, `az account` is designed to manage login profiles, instead of making ARM requests.

An "account" can also be a tenant account when `--allow-no-subscriptions` is used. Listing locations for a tenant ID would make no sense:

https://github.com/Azure/azure-cli/blob/bcb291ad55218dd8d7d582245d4fe97beb916309/src/azure-cli-core/azure/cli/core/commands/parameters.py#L28

I strongly recommend adding this `--include-extended-locations` argument to [`az account subscription list-location`](https://learn.microsoft.com/en-us/cli/azure/account/subscription?view=azure-cli-latest#az-account-subscription-list-location) from `account` extension, and deprecate `az account list-locations` gradually.

Moreover, I always feel `az account subscription` should be renamed as `az subscription` to make it aligned with the REST API.

Reply:
Another non-breaking-change solution would be to create a `subscription` command module and move `az account list-locations` there. Or, we move `az account list-locations` to `resource` module, as SDK client `SubscriptionClient` comes from `azure.mgmt.resource.subscriptions`:

https://github.com/Azure/azure-cli/blob/689cf6d53e46b5f71d7f49796a2d73c7b2226a77/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L60

Then we can consider

1. Move `account` extension to `subscription`/`resource` module
2. Rename `az account subscription` to `az subscription`

Or, introduce `az subscription list-locations` right now. After all, the REST API is called **Subscriptions - List Locations**: https://learn.microsoft.com/en-us/rest/api/resources/subscriptions/list-locations?tabs=HTTP

Reply:
Thanks for your good suggestion! Yes, I agree. Since we have committed to release this feature request in this sprint and this sprint is currently a non-breaking change window, I prefer to migrate it to resource module first
And this can follow the design style of other sub-RPs in resource module that were based on SDK subdirectories before